src/pages/TermsOfService.js

```diff
@@ -19,9 +19,16 @@ import * as Utils from 'src/libs/utils'
 const TermsOfServicePage = () => {
   const [busy, setBusy] = useState()
   const { isSignedIn, termsOfService } = authStore.get() // can't change while viewing this without causing it to unmount, so doesn't need to subscribe
-  const needToAccept = isSignedIn && !termsOfService.userAcceptedTos
-  const isGracePeriod = isSignedIn && termsOfService.isGracePeriodEnabled && !_.isUndefined(termsOfService.userAcceptedVersion)
+  const needToAccept = isSignedIn && termsOfService.showTosPopup
+  const userHasAcceptedPreviousTos = !_.isUndefined(termsOfService.userAcceptedVersion)
+  const canUserContinueUnderGracePeriod = isSignedIn && termsOfService.isGracePeriodEnabled && userHasAcceptedPreviousTos
   const [tosText, setTosText] = useState()
+  authStore.update(state => ({
```

Comment:
Just noticed a bunch of React warnings in the browser console that seem to stem from this. This shouldn't be updated in the render function. It needs to be in a useEffect hook or callback or something like that.

Reply:
Fixed in 767c48d64418ef9c8551c0a3e6c7d042dd15a516.